### PR TITLE
[ESIMD] Add tests for ESIMDOptimizeVecArgCallConv pass.

### PR DESCRIPTION
**Pull request by kbobrovs**
Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/vec_arg_call_conv_ext.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Intel GPU is not really required, but the test does not make sense for
+// others.
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl -Xclang -opaque-pointers -fsycl-device-only -Xclang -emit-llvm -o %t.comp.ll %s
```
Comment (v-klochkov) [inline SYCL/ESIMD/vec_arg_call_conv_ext.cpp:12]: You used '-opaque-pointers' here and '-opaque-ptr' for the 1st test above. Do both of spellings supported?

File: SYCL/ESIMD/vec_arg_call_conv_ext.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// CHECK:  call spir_func void @_Z22callee_void__noopt_opt{{.*}}(ptr addrspace(4) %{{.*}}, <8 x i32> %{{.*}})
+  return x;
+}
+
+//------------------------
+
+// Dummy main to satisfy linker.
+int main() {
+  return 0;
+}
```
Comment (v-klochkov) [inline SYCL/ESIMD/vec_arg_call_conv_ext.cpp]: nit: please add new line

File: SYCL/ESIMD/vadd_usm_opqptr.cpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//==---------------- vadd_usm_opqptr.cpp  - DPC++ ESIMD on-device test -----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl -Xclang -opaque-ptr %s -o %t.out
```
Comment (kbobrovs) [inline SYCL/ESIMD/vadd_usm_opqptr.cpp]:
```suggestion
// RUN: %clangxx -fsycl -Xclang -opaque-pointers %s -o %t.out
```

File: SYCL/ESIMD/vec_arg_call_conv_ext.cpp (hunk 1 above)
Comment (kbobrovs) [reply SYCL/ESIMD/vec_arg_call_conv_ext.cpp:12]: Thanks! No, just one, fixed.

**Review by v-klochkov — APPROVED** [review]

File: SYCL/ESIMD/vec_arg_call_conv_smoke.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+__attribute__((noinline))
+SYCL_EXTERNAL simd<float, 16> test__sret__fall_through__glob() SYCL_ESIMD_FUNCTION {
+  return callee__sret__param(V(GRF, 16, 0));
+// CHECK: %{{.*}} = call spir_func <16 x float> @_Z19callee__sret__param{{.*}}(<16 x float> %{{.*}})
+}
+
+// Dummy main to satisfy linker.
+int main() {
+  return 0;
+}
```
Comment (kbobrovs) [inline SYCL/ESIMD/vec_arg_call_conv_smoke.cpp]:
```suggestion
}

```

File: SYCL/ESIMD/vec_arg_call_conv_ext.cpp (hunk 2 above)
Comment (kbobrovs) [inline SYCL/ESIMD/vec_arg_call_conv_ext.cpp]:
```suggestion
}

```

**Comment by kbobrovs** [comment]
verification with https://github.com/intel/llvm/pull/6889 passed, #6889 is merged, merging.

**Pull request merged**